features/0434-outofband/README.md

```diff
@@ -0,0 +1,463 @@
+
+Two well known use cases for using the out of band protocol are:
+
+- A user on a web browser wants to execute a protocol (for example, to issue a verifiable credential) between the website operator's agent and a user's agent. To enable the execution of the protocol, the two agents must have a connection. An out-of-band message is used to trigger the creation of a new, or the reuse an existing, connection.
+- A website operator wants to request something from a user on their site where there is no (known) connection with the user.
+
+In both cases, there is only a single out of band protocol message sent. The message responding to the out of band message is a DIDComm message from an appropriate protocol.
+
+Note that the website-to-agent model is not the only such interaction enabled by the out of band protocol, and a QR code is not the only delivery mechanism for out of band messages. However, they are useful as examples of the purpose of the protocol.
+
+### Roles
+
+The out of band protocol has two roles: __sender__ and __receiver__.
+
+#### sender
+
+The agent that generates the out of band message and makes it available to the other party.
+
+#### receiver
+
+The agent that receives the out of band message and decides how to respond. There is no out of band protocol message with which the receiver will respond. Rather, if they respond, they will use a message from another protocol that the sender understands.
+
+### States
+
+The state machines for the sender and receiver are a bit odd for the out of band protocol because of it being a single message protocol with a response from a different protocol. In the following state machine diagrams we generically describe the response message from the *receiver* as being a DIDComm message without associating the message with a specific protocol.
+
+The *sender* state machine is as follows:
+
+[![sender state machine google doc](state-machine-sender.png)](https://docs.google.com/spreadsheets/d/1o1szGCMx6FzrxfAKV76IEYmeKRJqvV3gd1r3Qfv1Q9o/edit#gid=1176419697)
+
+Note the "optional" reference under the second event in the `await-response` state. That is to indicate that an out of band message might be a single use message with a transition to done, or reusable message (received by many receivers) with a transition back to `await-response`.
+
+The *receiver* state machine is as follows:
+
+[![receiver state machine google doc](state-machine-receiver.png)](https://docs.google.com/spreadsheets/d/1o1szGCMx6FzrxfAKV76IEYmeKRJqvV3gd1r3Qfv1Q9o/edit#gid=10874521)
+
+Worth noting is the first event of the `done` state, where the receiver may receive the message multiple times. This represents, for example, an agent returning to the same website and being greeted with the same QR code each time.
+
+### Messages
+
+There are two messages in the out of band protocol. They are both sent by the *sender*. Only one of the message types is ever used in an instance of the protocol.
```

Comment:
Only one of the message types is ever used in an instance of the protocol --> that's because these are two different protocols.

Reply:
Please propose exactly what change you want.  I don't see the benefit, but I'd like to get to the end of the discussion.

Reply:
Okay. Will do.

Reply:
Done. https://github.com/swcurran/aries-rfcs/pull/1